src/arch/io/network.cc

```diff
@@ -621,6 +621,430 @@ void linux_tcp_conn_t::on_event(int /* events */) {
     event_watcher->stop_watching_for_errors();
 }
 
+/* tls_conn_wrapper_t wraps a TLS connection and safely releases it in
+exceptional cases if it is not reclaimed before execution leaves its scope. */
+class tls_conn_wrapper_t {
+public:
+    tls_conn_wrapper_t(SSL *_conn) {
+        conn = _conn;
+    }
+
+    ~tls_conn_wrapper_t() {
+        SSL_free(conn);
+    }
+
+    // Transfer ownership of the connection to the caller.
+    void reclaim() {
```

Comment:
For consistency with other similar structures in our code and in the standard libraries, I suggest changing this function into the following:
```C++
// Transfer ownership of the connection to the caller.
SSL *release() {
    rassert(conn != nullptr);
    SSL *res = conn;
    conn = nullptr;
    return res;
}
```

This also fixes two other minor things:
1. Use `nullptr` instead of `NULL` (this is mostly a stylistic change)
2. Adds an assertion that the connection hasn't been released before. `rasserts` are enabled when compiling with `make DEBUG=1`.